File: Practice/30 Days of Code/Day 25 Running Time and Complexity/Day 25 Running Time and Complexity.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        while(n-- > 0) {
            int num = sc.nextInt();
            if( isPrime(num) ) System.out.println("Prime");
            else System.out.println("Not prime");
        }
    }
        private static boolean isPrime(int n) {
            if( n<2 ) return false;
            for( int i=2; i<=(int)Math.sqrt(n); i++)
                if( n%i == 0) return false;
            return true;
        }
}

File: Practice/30 Days of Code/Day 25 Running Time and Complexity/Day 25 Running Time and Complexity.js
function processData(input) {
    let result = input.split('\n').slice(1).map(isPrime);
    //Enter your code here
    function isPrime(input, i) {
        let messages = ['Not prime', 'Prime'];

        if (input == 1) { 
            return messages[0];
        }
        if (input == 2) {
            return messages[1];
        }
        if (input % 2 == 0) {
            return messages[0];
        } 
        
        for (i = 3; i <= Math.sqrt(input); i += 2) {
            if (input % i == 0) { 
                return messages[0];
            }
        } 
        return messages[1];
    }

    console.log(result.join("\n"));
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});
